pyfr/plugins/fluidforce.py

```diff
@@ -122,6 +160,32 @@ def __call__(self, intg):
             # Do the quadrature
             f += np.einsum('i...,ij,jik', qwts, p, norms)
 
+            if self.forcetype == 'viscous':
+                # Get operator and J^-T matrix
+                m4 = self._m4[etype]
+                rcpjact = self._rcpjact[etype, fidx]
+
+                # Transformed gradient at solution points
+                tduupts = np.dot(m4, uupts.reshape(nupts, -1))
+                tduupts = tduupts.reshape(ndims, nupts, nvars, -1)
+
+                # Physical gradient at solution points
+                duupts = np.einsum('ijkl,jkml->ikml', rcpjact, tduupts)
+
+                # Interpolate gradient to flux points
+                duupts = duupts.reshape(ndims, nupts, -1)
+                dufpts = np.zeros((ndims, nfpts) + duupts.shape[2:])
+                for i in range(ndims):
+                    dufpts[i] = np.dot(m0, duupts[i])
```

Comment:
`dufpts = np.something([np.dot(m0, u) for u in duupts])` or something similar to this should work.